packages/cdk/lib/cloudwatch-logs-management.ts

```diff
@@ -1,11 +1,214 @@
-import { join } from 'path';
+import { GuScheduledLambda } from '@guardian/cdk';
 import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
-import { GuStack } from '@guardian/cdk/lib/constructs/core';
+import { GuStack, GuStringParameter } from '@guardian/cdk/lib/constructs/core';
+import { GuLambdaFunction } from '@guardian/cdk/lib/constructs/lambda';
+import { GuS3Bucket } from '@guardian/cdk/lib/constructs/s3';
 import type { App } from 'aws-cdk-lib';
-import { CfnInclude } from 'aws-cdk-lib/cloudformation-include';
+import { Duration } from 'aws-cdk-lib';
+import { Schedule } from 'aws-cdk-lib/aws-events';
+import {
+	Effect,
+	ManagedPolicy,
+	PolicyStatement,
+	ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
+import { Runtime } from 'aws-cdk-lib/aws-lambda';
+
+export interface CloudwatchLogsManagementProps
+	extends Omit<GuStackProps, 'stage' | 'env'> {
+	retentionInDays?: number;
+	logShippingPrefixes?: string[];
+}
 
 export class CloudwatchLogsManagement extends GuStack {
-	constructor(scope: App, id: string, props: GuStackProps) {
-		super(scope, id, props);
+	constructor(scope: App, props: CloudwatchLogsManagementProps) {
+		const {
+			stack,
+			retentionInDays = 7,
+			logShippingPrefixes = ['/aws/lambda'],
+		} = props;
+
+		// The ID will become `CloudwatchLogsManagement-<STACK>`
+		const id = `${CloudwatchLogsManagement.prototype.constructor.name}-${stack}`;
+
+		super(scope, id, {
+			...props,
+			stack,
+			stage: 'PROD',
```

Comment:
Perhaps we should explain why this is hardcoded to `PROD` by slightly amending [this comment](https://github.com/guardian/cloudwatch-logs-management/blob/1f7271512ba39dd33fc94f67aad2940f5c64478a/template.yaml#L10-L13)?

Reply:
Done!